### PR TITLE
fix(world): prevent misconfigured delegations, allow unregistering [L-04]

### PR DESCRIPTION
**Pull request by yonadaaa**
Currently, we allow clearing delegations by passing empty calldata. However this could cause invalid delegations, so this PR performs full validation regardless of whether `initCallData` is empty. 

Delegations can now be explicitly unregistered by calling `unregisterDelegation`.

### CODE REVIEW

File: packages/world/src/modules/core/implementations/WorldRegistrationSystem.sol [hunk 1]
```diff
@@ -262,6 +262,13 @@ contract WorldRegistrationSystem is System, IWorldErrors {
     }
   }
 
+  function unregisterDelegation(address delegatee) public {
+    // Delete the delegation control contract address
+    UserDelegationControl.deleteRecord({ delegator: _msgSender(), delegatee: delegatee });
+
+    // TODO: clear the specific delegation control
```
Comment (yonadaaa) [inline packages/world/src/modules/core/implementations/WorldRegistrationSystem.sol:269]: Do we care about deleting the entries in the `DelegationControl`. If so should they implement some `deleteDelegation` function? 

Semantically, it's not obvious that calling `initDelegation` with empty calldata should "uninitialize" the delegation. For example, the control might increment some counter each time a delegation is registered.

**Comment by yonadaaa** [comment]
This PR is failing integration tests because `CoreSystem` is too big to deploy. This should be fixed in https://github.com/latticexyz/mud/pull/2046.

**Review by holic — APPROVED** [review]

File: packages/world/src/modules/core/implementations/WorldRegistrationSystem.sol [hunk 2]
```diff
@@ -260,6 +260,11 @@ contract WorldRegistrationSystem is System, IWorldErrors {
     }
   }
 
+  function unregisterDelegation(address delegatee) public {
```
Comment (holic) [inline packages/world/src/modules/core/implementations/WorldRegistrationSystem.sol:263]: need to make sure to add this to the core module function registration step

**Comment by holic** [comment]
Can we add a test while we're here, just to make sure we can unregister a delegation from the world?

**Comment by yonadaaa** [comment]
Added a test that you can unregister a delegation, then the call fails

**Review by holic — APPROVED** [review]

**Pull request merged**